NPE in the operator controller

```
E0819 11:21:22.985498       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 261 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x18b38a0, 0x2b22e40)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/runtime/runtime.go:48 +0x82
panic(0x18b38a0, 0x2b22e40)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).newJobServiceComponent(0xc0000a1db0, 0xc0000924e0)
        /workspace/controllers/harbor/provision.go:108 +0x31
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).newHarborCR(0xc0000a1db0, 0x0)
        /workspace/controllers/harbor/provision.go:52 +0x225
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).Provision(0xc0000a1db0, 0xc0003df720, 0x1ae5256, 0x8)
        /workspace/controllers/harbor/provision.go:21 +0x50
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).Reconcile(0xc0000a1db0, 0x1da8120, 0xc00004a200, 0xc00020e900)
        /workspace/controllers/harbor/harbor.go:38 +0x471
github.com/goharbor/harbor-cluster-operator/controllers.(*HarborClusterReconciler).Reconcile(0xc0005cc690, 0xc00059894a, 0x6, 0xc0006ac980, 0x11, 0xc0004a0800, 0xc0000bed88, 0x1d73240, 0xc0000bed80)
        /workspace/controllers/harborcluster_controller.go:147 +0x70d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0000bfdd0, 0x191d0e0, 0xc00037b4e0, 0x43e600)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:235 +0x27d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0000bfdd0, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:209 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0000bfdd0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0004a07c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:155 +0x5e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0004a07c0, 0x1d60e40, 0xc00031c7e0, 0x1, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004a07c0, 0x3b9aca00, 0x0, 0x1, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:133 +0xaa
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004a07c0, 0x3b9aca00, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:170 +0x431
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15ab041]

goroutine 261 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/runtime/runtime.go:55 +0x105
panic(0x18b38a0, 0x2b22e40)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).newJobServiceComponent(0xc0000a1db0, 0xc0000924e0)
        /workspace/controllers/harbor/provision.go:108 +0x31
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).newHarborCR(0xc0000a1db0, 0x0)
        /workspace/controllers/harbor/provision.go:52 +0x225
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).Provision(0xc0000a1db0, 0xc0003df720, 0x1ae5256, 0x8)
        /workspace/controllers/harbor/provision.go:21 +0x50
github.com/goharbor/harbor-cluster-operator/controllers/harbor.(*HarborReconciler).Reconcile(0xc0000a1db0, 0x1da8120, 0xc00004a200, 0xc00020e900)
        /workspace/controllers/harbor/harbor.go:38 +0x471
github.com/goharbor/harbor-cluster-operator/controllers.(*HarborClusterReconciler).Reconcile(0xc0005cc690, 0xc00059894a, 0x6, 0xc0006ac980, 0x11, 0xc0004a0800, 0xc0000bed88, 0x1d73240, 0xc0000bed80)
        /workspace/controllers/harborcluster_controller.go:147 +0x70d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0000bfdd0, 0x191d0e0, 0xc00037b4e0, 0x43e600)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:235 +0x27d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0000bfdd0, 0x0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:209 +0xcb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0000bfdd0)
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:188 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0004a07c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:155 +0x5e
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0004a07c0, 0x1d60e40, 0xc00031c7e0, 0x1, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:156 +0xa3
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0004a07c0, 0x3b9aca00, 0x0, 0x1, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:133 +0xaa
k8s.io/apimachinery/pkg/util/wait.Until(0xc0004a07c0, 0x3b9aca00, 0xc00008efc0)
        /go/pkg/mod/k8s.io/apimachinery@v0.19.0-rc.3/pkg/util/wait/wait.go:90 +0x4d
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func1
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.1-0.20200804124940-17eebbff0d48/pkg/internal/controller/controller.go:170 +0x431
```

--- a/controllers/harbor/provision.go
+++ b/controllers/harbor/provision.go
@@ -103,16 +103,19 @@ func (harbor *HarborReconciler) newRegistryComponent() *v1alpha1.RegistryCompone
 }
 
 func (harbor *HarborReconciler) newJobServiceComponent() *v1alpha1.JobServiceComponent {
-	return &v1alpha1.JobServiceComponent{
-		HarborDeployment: v1alpha1.HarborDeployment{
-			Replicas:         IntToInt32Ptr(harbor.HarborCluster.Spec.JobService.Replicas),
-			Image:            image.String(harbor.ImageGetter.JobServiceImage()),
-			NodeSelector:     nil,
-			ImagePullSecrets: harbor.getImagePullSecrets(),
-		},
-		RedisSecret: harbor.getCacheSecret(lcm.JobServiceSecretForCache),
-		WorkerCount: harbor.HarborCluster.Spec.JobService.WorkerCount,
+	if harbor.HarborCluster.Spec.JobService != nil {
+		return &v1alpha1.JobServiceComponent{
+			HarborDeployment: v1alpha1.HarborDeployment{
+				Replicas:         IntToInt32Ptr(harbor.HarborCluster.Spec.JobService.Replicas),
+				Image:            image.String(harbor.ImageGetter.JobServiceImage()),
+				NodeSelector:     nil,
+				ImagePullSecrets: harbor.getImagePullSecrets(),
+			},
+			RedisSecret: harbor.getCacheSecret(lcm.JobServiceSecretForCache),
+			WorkerCount: harbor.HarborCluster.Spec.JobService.WorkerCount,
+		}
 	}
+	return nil
 }
 
 func (harbor *HarborReconciler) newChartMuseumComponentIfNecessary() *v1alpha1.ChartMuseumComponent {

